File: 2nd sem/GA/Homework 1/python/Externals/read_write.py


def read_graph_from_file(graph, file_path):

    with open(file_path) as fp:
        line = fp.readline()
        elements = line.split()
        if len(elements) != 2:
            raise ValueError("Incorrect file format! First line should only contain 2 integers!\n")

        nodes = int(elements[0])
        edges = int(elements[1])

        for i in range(nodes):
            graph.add_node(i)

        for i in range(edges):
            line = fp.readline()
            elements = line.split()

            if len(elements) != 3:
                raise ValueError("Incorrect file format!\n")

            graph.add_edge(int(elements[0]), int(elements[1]), int(elements[2]))


def write_graph_in_file(graph, file_path):
    g = open(file_path, "w")

    g.write(str(graph.get_number_of_nodes()) + " " + str(graph.get_number_of_edges()) + "\n")

    for source, target, cost in graph.get_edges():
        g.write(str(source) + " " + str(target) + " " + str(cost) + "\n")

    g.close()

File: 2nd sem/GA/Homework 1/c++/Externals/read_write.cpp
//
// Created by susci on 3/28/2021.
//

#include "../Graphs/DirectedGraph.h"
#include <string>
#include <fstream>
//
//void read_graph_from_file_DA(DirectedGraph& graph)
//{
//    std::ifstream f("graph.in");
//    int source, target, cost;
//    int number_nodes, number_edges;
//
//    f >> number_nodes >> number_edges;
//
//    for (int i = 0; i < number_nodes; i++)
//        graph.add_node(i);
//
//    for (int i = 0; i < number_edges; i++)
//    {
//        f >> source >> target >> cost;
//        graph.add_edge(source, target, cost);
//    }
//    f.close();
//}

//void write_graph_in_file(DirectedGraph& graph, std::string& filename)
//{
//    std::ofstream g(filename);
//
//    int nodes = graph.get_number_nodes();
//    int edges = graph.get_number_edges();
//
//    g << nodes << " " << edges << '\n';
//
//    std::vector <std::tuple<int, int, int> > all_edges = graph.parse_edges();
//
////    for (auto & all_edge : all_edges)
////    {
////        g << std::get<0>(all_edge) << " " << std::get<1 >(all_edge) << " " << std::get<2>(all_edge) << "\n";
////    }
//
//}